test/lib/processors/manifestEnricher.js

```diff
@@ -0,0 +1,1647 @@
+	t.is(t.context.logVerboseSpy.callCount, 1, "1 verbose should be logged");
+	t.is(t.context.logVerboseSpy.getCall(0).args[0],
+		"manifest.json: _version is lower than 1.21.0 so no supportedLocales can be generated");
+	t.true(t.context.logWarnSpy.notCalled, "No warning should be logged");
+	t.true(t.context.logErrorSpy.notCalled, "No errors should be logged");
+});
+
+test.serial("Application: sap.ui5/models: " +
+	"Do not generate supportedLocales when bundleUrl pointing to a location outside the current project",
+async (t) => {
+	t.plan(5);
+	const {manifestEnricher} = t.context;
+	const input = JSON.stringify({
+		"_version": "1.58.0",
+		"sap.app": {
+			"id": "sap.ui.demo.app",
+			"type": "application"
+		},
+		"sap.ui5": {
+			"models": {
+				"i18n": {
+					"type": "sap.ui.model.resource.ResourceModel",
+					"settings": {
+						"bundleUrl": "../../myapp2/i18n/i18n.properties"
+					}
+				}
+			}
+		}
+	}, null, 2);
+
+	const resource = createResource("/resources/sap/ui/demo/app/manifest.json", true, input,
+		(actual) => t.fail("setString should never be called because resource should not be changed"));
+
+	const processedResources = await manifestEnricher({
+		resources: [resource],
+		fs: {
+			readdir() {
+				t.fail("fs.readDir should not be called because generation of supported locales is disablaed");
+			},
+			stat: ((fsPath, callback) => callback(null, undefined))
+		}
+	});
+
+	t.deepEqual(processedResources, [undefined], "No resource is returned, because it is not changed");
+	t.is(t.context.logVerboseSpy.callCount, 1, "1 verbose should be logged");
+	t.is(t.context.logVerboseSpy.getCall(0).args[0],
+		"manifest.json: bundleUrl '../../myapp2/i18n/i18n.properties' contains a relative path, " +
+		"no supportedLocales are generated");
+	t.true(t.context.logWarnSpy.notCalled, "No warning should be logged");
+	t.true(t.context.logErrorSpy.notCalled, "No errors should be logged");
+});
+
+test.serial("Application: sap.ui5/models: " +
+	"Do not generate supportedLocales when bundleUrl pointing to a location inside the current project",
+async (t) => {
+	t.plan(5);
+	const {manifestEnricher} = t.context;
+	const input = JSON.stringify({
+		"_version": "1.58.0",
+		"sap.app": {
+			"id": "sap.ui.demo.app",
+			"type": "application"
+		},
+		"sap.ui5": {
+			"models": {
+				"i18n": {
+					"type": "sap.ui.model.resource.ResourceModel",
+					"settings": {
+						"bundleUrl": "../i18n/i18n.properties",
+						"fallbackLocale": "de"
+					}
+				}
+			}
+		}
+	}, null, 2);
+
+	const expected = JSON.stringify({
+		"_version": "1.58.0",
+		"sap.app": {
+			"id": "sap.ui.demo.app",
+			"type": "application"
+		},
+		"sap.ui5": {
+			"models": {
+				"i18n": {
+					"type": "sap.ui.model.resource.ResourceModel",
+					"settings": {
+						"bundleUrl": "../i18n/i18n.properties",
+						"fallbackLocale": "de",
+						"supportedLocales": ["de", "en"]
```

Comment:
This expectation doesn't match with the description of the test. I think this is unexpected and the test should actually expect that the manifest is unchanged.

Reply:
Test has been updated accordingly